src/main/java/Team/TeamLeader.java

```diff
@@ -0,0 +1,23 @@
+package Team;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TeamLeader extends TeamMember {
+    public List<TeamMember> teamMembers;
+    public TeamLeader(String name, String teamName) {
+        super(name, teamName);
+        this.teamMembers = new ArrayList<TeamMember>();
+        this.teamMembers.add(this);
+    }
+
+    public void addTeamMember(TeamMember newTeamMember){
```

Comment:
DIP- classes should be based on abstractions (such as interfaces) and not on specific implementations